lib/neo4j/shared/enum.rb

```diff
@@ -0,0 +1,120 @@
+require 'active_support/core_ext/hash/except'
+require 'active_support/core_ext/hash/slice'
+
+module Neo4j::Shared
+  module Enum
+    extend ActiveSupport::Concern
+
+    class ConflictingEnumMethodError < Neo4j::Error; end
+
+    module ClassMethods
+      attr_reader :neo4j_enum_data
+
+      def enum(parameters = {})
+        options, parameters = *split_options_and_parameters(parameters)
```

Comment:
Uhm, in ActiveRecord enum is using this convention:

Parameters are all key-values.

Keys that starts with `_` (`_prefix`, `_suffix`, ecc...) are options, the other are the name of the enum properties.

Check here:
http://edgeapi.rubyonrails.org/classes/ActiveRecord/Enum.html

I just tried to replicate it...

If you prefer, I can change this:

```ruby
enum key: [:list, :of, :keys], _option: true
```
to this:

```ruby
enum :key, [:list, :of, :keys], option: true
```